draft-jholland-quic-multicast.md

```diff
@@ -221,6 +221,34 @@ If a server or client somehow still detect a stateless reset for a channel, they
 
 # New Frames
 
+## MC_RESERVE_CHANNEL_IDS {#reserve-channel-ids-frame}
+
+If a server learns that a client supports multicast through its transport parameters, it SHOULD immediately send a MC_RESERVE_CHANNEL_IDS frame (type=TBD-00) that includes all channel IDs that might be used by its multicast channels.
+This is done to prevent a client from using those connection IDs for the unicast connection, which could cause a delay when attempting to join a channel as they would first have to be retired for the unicast connection.
+
+MC_RESERVE_CHANNEL_IDS frames are formatted as shown in {{fig-mc-channel-reserve-ids}}.
+
+~~~
+MC_RESERVE_CHANNEL_IDS Frame {
+  Type (i) = TBD-00 (experiments use 0xff3e800),
+  Length (i),
+  Channel ID Lengths (8) ...,
```

Comment:
I think a list of { Length, ChannelID } structures is probably better.  There's an existing [WriteLengthPrefixedConnectionId](https://github.com/GrumpyOldTroll/quiche/blob/jake-demo-branch/quic/core/quic_data_writer.h#L88) function that can be used if it's structured that way.

Reply:
Right, good point about the fragmentation and list format. I guess another option would be to just have a range of reserved channels, so something like

`Length start (8),
Start channel ID (8..160),
Length end (8),
End channel ID (8..160),'

I guess the question is if there is any serious reason why a server would want to have several entirely separate ranges of channel IDs? Or maybe you just allow to send several of those frames, in cases where it is really required?